index.bs

```diff
@@ -316,57 +316,61 @@ Interface {#interface}
+  DOMString architecture = "";
+  DOMString model = "";
+};
+
 [Exposed=Window]
 interface NavigatorUAData {
-  readonly attribute FrozenArray&lt;NavigatorUABrandVersionDict&gt; brand;
+  readonly attribute FrozenArray&lt;NavigatorUABrandVersion&gt; UASet;
   readonly attribute boolean mobile;
-  Promise&lt;DOMString&gt; getPlatform();
-  Promise&lt;SOMString&gt; getPlatformVersion();
-  Promise&lt;DOMString&gt; getArchitecture();
-  Promise&lt;DOMString&gt; getModel();
+  Promise&lt;UADataValues&gt; getHighEntropyValues(optional UADataOptions options);
 };
 
 interface mixin NavigatorUA {
-  [SecureContext] NavigatorUAData getUserAgent();
+  [SecureContext] readonly attribute NavigatorUAData userAgentData;
 };
 Navigator includes NavigatorUA;
 
 </pre>
 
 Processing model {#processing}
 --------------
-<dfn method for="NavigatorUA"><code>getUserAgent()</code></dfn> method MUST run these steps:
+On getting, the {{NavigatorUAData/UASet}} attribute must return the set of {{NavigatorUABrandVersion}} objects, initialized with the user agent's [=user agent/brand=] and [=user agent/significant version=].
 
-1. Let |p| be a [=a new promise=].
+On getting, the {{NavigatorUAData/mobile}} attribute must return the user agent's [=user agent/mobileness=].
 
-2.  Run the following steps [=in parallel=]:
+The <dfn method for="NavigatorUA"><code>getHighEntropyValues()</code></dfn> method, optionally called with {{UADataOptions}} |options|, MUST run these steps:
 
-    1.  Let |UAData| be a new {{NavigatorUAData}} object whose values are initialized as follows:
+1. Let |p| be a [=a new promise=].
 
-        :   {{NavigatorUAData/brand}}
-        ::  The user agent's [=user agent/brand=].
-        :   {{NavigatorUAData/getPlatform}}
-        ::  The user agent's [=user agent/platform brand=].
-        :   {{NavigatorUAData/getPlatformVersion}}
-        ::  The user agent's [=user agent/platform version=].
-        :   {{NavigatorUAData/getArchitecture}}
-        ::  The user agent's [=user agent/platform architecture=].
-        :   {{NavigatorUAData/getModel}}
-        ::  The user agent's [=user agent/model=].
-        :   {{NavigatorUAData/mobile}}
-        ::  The user agent's [=user agent/mobileness=].
+2.  Run the following steps [=in parallel=]:
 
-    2.  [=Resolve=] |p| with |UAData|.
+    1. If |options| is the empty object, let |options| be a new {{UADataOptions}}.
+    2. Let |UAData| be a new {{UADataValues}}.
+    3. If |options.platform| is true, let |UAData.platform| be the the user agent's [=user agent/platform brand=].
+    4. If |options.platformVersion| is true, let |UAData.platformVersion| be the the user agent's [=user agent/platform version=].
+    5. If |options.architecture| is true, let |UAData.architecture| be the the user agent's [=user agent/platform architecture=].
+    6. If |options.model| is true, let |UAData.model| be the the user agent's [=user agent/model=].
+    7. [=Resolve=] |p| with |UAData|.
```

Comment:
You need to queue a task on a specific task source to resolve this promise.

Reply:
Queued the task. I'm not sure which queue from the [generic task queues](https://html.spec.whatwg.org/multipage/webappapis.html#generic-task-sources) would be the appropriate one. Added an ISSUE to that effect